fetch-in-react-practice/client/src/components/Todos.jsx

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+export default function Todos() {
+  const [todos, setTodos] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function fetchAllTodos() {
+      try {
+        const response = await fetch('/api/todos');
+        if (!response.ok)
+          throw new Error(`Network response was not OK:', ${response.status}`);
+        const todos = await response.json();
+        setTodos(todos);
+      } catch (error) {
+        console.error(error.message);
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchAllTodos();
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo) {
+    try {
+      const response = await fetch('/api/todos', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(newTodo),
+      });
+      if (!response.ok)
+        throw new Error(`Network response was not OK:', ${response.status}`);
+      const newItemTodo = await response.json();
+      setTodos(todos.concat(newItemTodo));
+    } catch (error) {
+      console.error(error.message);
+      setError(error);
+    }
+  }
+
+  /* Implement toggleCompleted to toggle the completed state of a todo. Hints are at the bottom of the file. */
+  async function toggleCompleted(todoId) {
+    const todoObjIndex = todos.findIndex((x) => x.todoId === todoId);
```

Comment:
It's easier to use `.find`, which returns the actual item (or `undefined` if it couldn't find one).